File: src/main/java/ru/melonhell/bulletphysics/init/CommandBeanPostProcessor.java
package ru.melonhell.bulletphysics.init;

import co.aikar.commands.BaseCommand;
import co.aikar.commands.PaperCommandManager;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.BeansException;
import org.springframework.beans.factory.config.BeanPostProcessor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class CommandBeanPostProcessor implements BeanPostProcessor {
    private final PaperCommandManager commandManager;

    @Override
    public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
        if (bean instanceof BaseCommand) commandManager.registerCommand((BaseCommand) bean);
        return bean;
    }
}

File: src/main/java/ru/melonhell/bulletphysics/init/CommandBeanPostProcessor.kt
package ru.melonhell.bulletphysics.init

import co.aikar.commands.BaseCommand
import co.aikar.commands.PaperCommandManager
import org.springframework.beans.BeansException
import org.springframework.beans.factory.config.BeanPostProcessor
import org.springframework.stereotype.Component

@Component
class CommandBeanPostProcessor(private val commandManager: PaperCommandManager) : BeanPostProcessor {
    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any {
        if (bean is BaseCommand) commandManager.registerCommand(bean)
        return bean
    }
}